ddp-workspace/projects/study-builder-ui/src/app/components/picklist-question-block/picklist-question-block.component.ts

```diff
@@ -0,0 +1,77 @@
+import { ChangeDetectionStrategy, Component, Input, OnInit } from '@angular/core';
+import { QuestionBlockDef } from '../../model/core/questionBlockDef';
+import { ActivityPicklistOption, ActivityPicklistQuestionBlock } from 'ddp-sdk';
+import { Observable } from 'rxjs';
+import { ConfigurationService } from '../../configuration.service';
+import { filter, map, tap } from 'rxjs/operators';
+import { SimpleTemplate } from '../../model/core-extended/simpleTemplate';
+import { PicklistQuestionDef } from '../../model/core/picklistQuestionDef';
+import { PicklistOptionDef } from '../../model/core/picklistOptionDef';
+import { StudyConfigObjectFactory } from '../../model/core-extended/studyConfigObjectFactory';
+
+@Component({
+    selector: 'app-picklist-question-block',
+    templateUrl: './picklist-question-block.component.html',
+    styleUrls: ['./picklist-question-block.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class PicklistQuestionBlockComponent implements OnInit {
+    @Input()
+    definitionBlock$: Observable<QuestionBlockDef<PicklistQuestionDef>>;
+    angularClientBlock$: Observable<ActivityPicklistQuestionBlock>;
+
+    private factory: StudyConfigObjectFactory;
+    constructor(private config: ConfigurationService) {
+        this.factory = new StudyConfigObjectFactory(config);
+    }
+
+    ngOnInit(): void {
+        this.angularClientBlock$ = this.definitionBlock$.pipe(
+            tap(defBlock => console.log('getting defblock: %o', defBlock)),
+            filter(block => !!block),
+            map(defBlock => this.buildFromDef(defBlock)));
+    }
+
+    private buildFromDef(defBlock: QuestionBlockDef<PicklistQuestionDef>): ActivityPicklistQuestionBlock {
+        // todo replace mock options with custom options once options control would be added
+        const text1 = new SimpleTemplate(this.factory.createBlankTemplate());
```

Comment:
Suggest extracting  options set logic (line 36-42) into a separate method, 
like `const options = this.getOptions()`.
What do you think?

Reply:
I'm going to remove this mock logic in the next PR :)